scrapy/http/request/__init__.py

```diff
@@ -55,6 +58,22 @@ def meta(self):
             self._meta = {}
         return self._meta
 
+    def _get_params(self):
+        return self._params
+
+    def _set_params(self, params):
+        if params is None:
+            self._params = None
+            return
+        if any([isinstance(params, dict), isinstance(params, list)]):
+            qs = urlencode(params, doseq=True)
+            self._set_url(self.url + ('&' if '?' in self.url else '?') + qs)
+            self._params = qs
+        else:
+            raise TypeError('Params must be dict or a list of 2-tuples, got {}'.format(type(params)))
+
+    params = property(_get_params, obsolete_setter(_set_params, 'params'))
```

Comment:
`obsolete_setter` seems meant for properties that use to be writable and are now read-only, which is not the case.

I think you can just omit the setter part altogether.

Reply:
Setter part removed.